src/core/collector.ml

```diff
@@ -71,3 +75,27 @@ module type S = sig
   val shutdown : unit -> unit
   (** Shutdown collector, possibly waiting for it to finish sending data. *)
 end
+
+(** Default implementation that does nothing. Collectors that wish to
+    be more resilient in the face of futures additions to {!S}
+    can start with [include Trace.Collector.Default]. *)
+module Default : S = struct
+  let _fail what =
+    failwith (Printf.sprintf "not implemented: Trace.Collector.%s" what)
```

Comment:
Perhaps it's a good idea to mirror my implementation, and require collectors expose a string-ish `name` that can be printed in debugging contexts like this one?

https://github.com/ELLIOTTCABLE/ocaml-ambient-context/blob/30073325fe8d168ce44ac06b8b338d5b4875b02c/lib/types.ml

Reply:
In fact, perhaps this should even be abstracted — in additional to `message`, add `internal_message`, which `ocaml-trace` itself will call in erroneous situations? This allows collectors to self-collect.